src/Uno.UI/UI/Xaml/DependencyPropertyDetailsCollection.cs

```diff
@@ -32,21 +37,33 @@ public DependencyPropertyDetailsCollection(Type ownerType, ManagedWeakReference
 
 			var propertiesForType = DependencyProperty.GetPropertiesForType(ownerType);
 
-			var entries = new PropertyEntry[propertiesForType.Length];
+			var entries = _pool.Rent(propertiesForType.Length);
 
 			for (int i = 0; i < propertiesForType.Length; i++)
 			{
-				entries[i].Id = propertiesForType[i].UniqueId;
+				ref var entry = ref entries[i];
+				entry.Id = propertiesForType[i].UniqueId;
+				entry.Details = null;
 			}
 
 			// Entries are pre-sorted by the DependencyProperty.GetPropertiesForType method
-			AssignEntries(entries, sort: false);
+			AssignEntries(entries, propertiesForType.Length, sort: false);
 
 			// Prefetch known properties for faster access
 			DataContextPropertyDetails = GetPropertyDetails(dataContextProperty);
 			TemplatedParentPropertyDetails = GetPropertyDetails(templatedParentProperty);
 		}
 
+		internal void Dispose()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Either implement 'IDisposable.Dispose', or totally rename this method to prevent confusion.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5247895)